src/IntervalArith/Extended.purs

```diff
@@ -0,0 +1,77 @@
+{-
+  Based on CDAR.Extended by Jens Blanck (https://github.com/jensblanck/cdar).
+-}
+{- |The Extended module allows real-valued numeric data types to be extended by
+   positive and negative infinity.
+-}
+module IntervalArith.Extended (Extended(..)) where
+
+import Prelude
+import Control.Apply (lift2)
+
+-- |Extended numbers are either finite numbers or one of the two infinities.
+data Extended a
+  = PosInf
+  | NegInf
+  | Finite a
+
+derive instance eqExtended :: (Eq a) => Eq (Extended a)
+
+derive instance functorExtended :: Functor Extended
+
+instance showExtended :: Show a => Show (Extended a) where
+  show PosInf = "PosInf"
+  show NegInf = "NegInf"
+  show (Finite a) = "Finite (" <> show a <> ")"
+
+instance applyExtended :: Apply Extended where
+  apply (Finite f) (Finite x) = Finite $ f x
+  apply (Finite _) PosInf = PosInf
+  apply (Finite _) NegInf = NegInf
+  apply PosInf _ = PosInf
+  apply NegInf _ = NegInf
+
+instance applicativeExtended :: Applicative Extended where
+  pure = Finite
+
+instance bindExtended :: Bind Extended where
+  bind (Finite x) f = f x
+  bind PosInf _ = PosInf
+  bind NegInf _ = NegInf
+
+instance monadExtended :: Monad Extended
+
+instance ordExtended :: Ord a => Ord (Extended a) where
+  compare PosInf PosInf = EQ
+  compare NegInf NegInf = EQ
+  compare _ PosInf = LT
+  compare NegInf _ = LT
+  compare PosInf _ = GT
+  compare _ NegInf = GT
+  compare (Finite a) (Finite b) = compare a b
+
+instance semiringExtended :: (Ord a, Semiring a) => Semiring (Extended a) where
+  zero = pure zero
+  one = pure one
+  -- PosInf + NegInf should be undefined, but here it is the first argument
+  add = lift2 add
+  -- 0 * ???Inf should be undefined, but here it is PosInf
+  mul a PosInf = if a < zero then NegInf else PosInf
+  mul PosInf a = if a < zero then NegInf else PosInf
+  mul a NegInf = if a < zero then PosInf else NegInf
+  mul NegInf a = if a < zero then PosInf else NegInf
+  mul a b = (*) <$> a <*> b
+
+instance ringExtended :: (Ord a, Ring a) => Ring (Extended a) where
+  sub = lift2 sub
+
+instance commutativeringExtended :: (Ord a, CommutativeRing a) => CommutativeRing (Extended a)
+
+--     abs = fmap abs
```

Comment:
Removed zombie code

Reply:
Thank you for looking into this.  I am confused by this request though.  Firstly, GitHub shows four lines being added but I suspect that it is a confusing way of telling me that the four lines are being deleted.  I will assume this is the case. 

I do not think that the four lines should be deleted. The CommutativeRing instance is important, even though this class does not add any operations.  It is there to declare that the multiplication for this type is commutative.  

The commented out definition of abs is part of what I copied from the Haskell version ready for conversion to PureScript when we need it.  I guess this and the other 7 commented out lines could be deleted and brought back from the Haskell version again if you prefer.  What do you think?

Reply:
It is not good practice to have commented out code in a codebase. The main reason is that when refactoring or new work is undertaken, the "zombie" code has to be refactored as well which is a waste time as it is not used. That or it becomes outdated and no longer works when it is needed.

Reply:
Makes sense.  I have removed/finished the Zombie code.  Thank you!